tests/test_serialize.py

```diff
@@ -394,3 +394,40 @@ class MyModel(BaseModel):
             @classmethod
             def _serialize(self, x, y, z):
                 return self
+
+
+def test_field_multiple_serializer():
+    m = (
+        'Duplicate field serializer function '
+        '"tests.test_serialize::test_field_multiple_serializer.<locals>.MyModel.serializer2" for field "x"'
+    )
+    with pytest.raises(TypeError, match=m):
+
+        class MyModel(BaseModel):
+            x: int
+            y: int
+
+            @field_serializer('x', 'y', json_return_type='str')
+            def serializer1(v, _info):
+                return f'{v:,}'
+
+            @field_serializer('x', json_return_type='str')
+            def serializer2(v, _info):
+                return v
+
+
+def test_field_multiple_serializer_subclass():
+    class MyModel(BaseModel):
+        x: int
+
+        @field_serializer('x', json_return_type='str')
+        def customise_x_serialisation(v, _info):
+            return f'{v:,}'
+
+    class MySubModel(MyModel):
+        @field_serializer('x', json_return_type='str')
+        def customise_x_serialisation(v, _info):
+            return f'{v}'
```

Comment:
What happens if the function name is not the same?

Reply:
same behavior. it checks based on field_name and class 
I will change the second function name